### DISCUSSION TITLE
Does Repartition rule works in balista

### DISCUSSION THREAD

**Discussion by mustafasrepo** [apache/datafusion#7178] [Q&A] [answered]
In the comment for repartition rule. There is a comment saying that this rule doesn't work in balista.
See relevant [section](https://github.com/apache/arrow-datafusion/blob/1e9816b07edd8e73bb45ababa20a7c22de492d96/datafusion/core/src/physical_optimizer/optimizer.rs#L81). 

I have examined balista code a bit, However it seems that this is not valid comment. Is this so, or am I missing something? @waynexia . Are you familiar with current balista side.

**Comment by alamb** [comment]
Maybe @thinkharderdev  or @yahoNanJing  have ideas as well

**Reply by thinkharderdev** [reply, marked as answer]
I don't think this is a valid comment. AFAICT ballista does use this optimizer rule (or rather doesn't disable it). Maybe the comment was originally added since the cost of adding another repartition in the distributed plan is much higher since it adds another shuffle stage.

**Reply by mustafasrepo** [reply]
Thanks @thinkharderdev .
